libs/parallelism/algorithms/include/hpx/parallel/algorithms/for_each.hpp

```diff
@@ -560,30 +487,70 @@ namespace hpx { namespace parallel { inline namespace v1 {
             }
         };
 
-        ///////////////////////////////////////////////////////////////////////
-        // non-segmented implementation
-        template <typename ExPolicy, typename FwdIterB, typename FwdIterE,
+        /// Non Segmented implementation
+        template <typename ExPolicy, typename FwdIter, typename Size,
             typename F, typename Proj>
-        inline typename util::detail::algorithm_result<ExPolicy, FwdIterB>::type
-        for_each_(ExPolicy&& policy, FwdIterB first, FwdIterE last, F&& f,
+        typename util::detail::algorithm_result<ExPolicy, FwdIter>::type
+        for_each_n_(ExPolicy&& policy, FwdIter first, Size count, F&& f,
```

Comment:
Just to check: this `for_each_n_` and the declaration for the segmented one below are only kept around for the deprecated `hpx::parallel::for_each_n`? Although I see you changed `hpx::parallel::for_each` to use `for_each::call` directly. Can we do the same here so that we can just remove these `for_each_n_`s? Am I missing some other place where they're required?

Reply:
My plan was to make this ready first and then to look through all other algorithms. `for_each_n` would be the next, I guess.